blockchain/standalone/treasury_test.go

```diff
@@ -9,19 +9,106 @@ import (
+	mainNetTVI = 288
+
+	// mainNetTVIMul is the treasury vote interval multiplier for mainnet.
+	mainNetTVIMul = 12
+)
+
+// TestCalcTSpendWindow ensures that the function that calculates the start and
+// end of a treasury spend voting window returns the expected results including
+// error conditions.
+func TestCalcTSpendWindow(t *testing.T) {
+	tests := []struct {
+		name      string // test description
+		expiry    uint32 // expiry to calculate the window for
+		tvi       uint64 // treasury vote interval
+		tvimul    uint64 // treasury vote interval multiplier
+		err       error  // expected error
+		wantStart uint32 // expected start result
+		wantEnd   uint32 // expected end result
+	}{{
+		name:      "zero is not a valid expiry",
+		expiry:    0,
+		tvi:       mainNetTVI,
+		tvimul:    mainNetTVIMul,
+		err:       ErrInvalidTSpendExpiry,
+		wantStart: 0,
+		wantEnd:   0,
+	}, {
+		name:      "min required expiry - 1",
+		expiry:    mainNetTVI*mainNetTVIMul + 1,
+		tvi:       mainNetTVI,
+		tvimul:    mainNetTVIMul,
+		err:       ErrInvalidTSpendExpiry,
+		wantStart: 0,
+		wantEnd:   0,
+	}, {
+		name:      "not a TVI + 2",
+		expiry:    mainNetTVI*mainNetTVIMul + 3,
+		tvi:       mainNetTVI,
+		tvimul:    mainNetTVIMul,
+		err:       ErrInvalidTSpendExpiry,
+		wantStart: 0,
+		wantEnd:   0,
+	}, {
+		name:      "5 is not a valid start or end for a tvi 11, mul 3",
+		expiry:    5,
+		tvi:       11,
+		tvimul:    3,
+		err:       ErrInvalidTSpendExpiry,
+		wantStart: 0,
+		wantEnd:   0,
+	}, {
+		name:      "first possible valid mainnet params",
+		expiry:    mainNetTVI*mainNetTVIMul + 2,
+		tvi:       mainNetTVI,
+		tvimul:    mainNetTVIMul,
+		err:       nil,
+		wantStart: 0,
+		wantEnd:   0,
+	}, {
+		name:      "second possible valid mainnet params",
+		expiry:    mainNetTVI*mainNetTVIMul*2 + 2,
+		tvi:       mainNetTVI,
+		tvimul:    mainNetTVIMul,
+		err:       nil,
+		wantStart: mainNetTVI * mainNetTVIMul,
+		wantEnd:   0,
+	}, {
+		name:      "5186 for tvi 288, mul 7 is window [3168, 5184)",
+		expiry:    5186,
+		tvi:       288,
+		tvimul:    7,
+		err:       nil,
+		wantStart: 5186 - 288*7 - 2,
+		wantEnd:   5186 - 2,
+	}}
 
-	// 5 is not a valid end for a tvi of 11.
-	_, err = CalculateTSpendWindowEnd(5, 11)
-	if !errors.Is(err, ErrTSpendEndInvalidExpiry) {
-		t.Fatalf("expected %v got %v",
-			ErrTSpendEndInvalidExpiry, err)
+	for _, test := range tests {
+		// Calculate result and ensure the expected error is produced.
+		start, end, err := CalcTSpendWindow(test.expiry, test.tvi, test.tvimul)
+		if !errors.Is(err, test.err) {
+			t.Errorf("%q: unexpected error -- got %v, want %v", test.name, err,
+				test.err)
+			continue
+		}
+
+		// Ensure the expected start value is calculated.
+		if start != test.wantStart {
+			t.Errorf("%q: unexpected start val -- got %v, want %v", test.name,
+				start, test.wantStart)
+			continue
+		}
+
+		// Ensure the expected end value is calculated.
+		if start != test.wantStart {
```

Comment:
Looks like a copy/paste error from above
```suggestion
		if end != test.wantEnd {
```

Reply:
Nice catch!  I see I had a couple of uncommitted changes locally in this test for the end values too.  Fixed.